src_c/transform.c

```diff
@@ -570,6 +609,110 @@ scale_to(pgSurfaceObject *srcobj, pgSurfaceObject *dstobj, int width,
     return retsurf;
 }
 
+SDL_Surface *
+grayscale(pgSurfaceObject *srcobj, pgSurfaceObject *dstobj)
+{
+    SDL_Surface *src = pgSurface_AsSurface(srcobj);
+    SDL_Surface *newsurf;
```

Comment:
Should check if the surface has been initialized before proceeding. Related to PR https://github.com/pygame/pygame/pull/3525 and issue https://github.com/pygame/pygame/issues/3519